.github/workflows/ci-build.yaml

```
@@ -401,6 +401,14 @@ jobs:
       - run: yarn --cwd ui test
       - run: yarn --cwd ui lint
       - run: yarn --cwd ui deduplicate
+      # Deduplicate UI deps for dependabot PRs
+      - if: github.actor == 'dependabot[bot]'
+        run: |
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+          git add ui/yarn.lock
+          git commit -s -m 'chore: deduplicate yarn.lock'
+          git push
```

Welp, now the `git commit` passes with an `--allow-empty`, but the `git push` line is failing. From [failing CI](https://github.com/argoproj/argo-workflows/actions/runs/8562226554/job/23465109452?pr=12891) in #12891:

```
Run git config --global user.name 'github-actions[bot]'
  git config --global user.name 'github-actions[bot]'
  git config --global user.email 'github-actions[bot]@users.noreply.github.com'
  git add ui/yarn.lock
  git commit -s --allow-empty -m 'chore: deduplicate yarn.lock'
  git push
  shell: /usr/bin/bash -e {0}
  env:
    NODE_OPTIONS: --max-old-space-size=4096
[detached HEAD 4ab78f5] chore: deduplicate yarn.lock
fatal: You are not currently on a branch.
To push the history leading to the current (detached HEAD)
state now, use

    git push origin HEAD:<name-of-remote-branch>

Error: Process completed with exit code 1[2](https://github.com/argoproj/argo-workflows/actions/runs/8561962725/job/23464600770?pr=12880#step:9:2)8.
```

---

We should be able to use [`$GITHUB_HEAD_REF`](https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables) as the branch name (per the docs, it is set for PRs, and dependabot is always a PR), so something like:

```suggestion
          git push origin HEAD:$GITHUB_HEAD_REF
```

but honestly this needs some more testing given all the errors so far. I would suggest trying this on a PR on a fork and skipping the dependabot `actor` check for testing purposes

---

I'm testing this in https://github.com/agilgur5/argo-workflows/pull/3. I had to add `permissions.contents: write` as well (which isn't great, but necessary, with no real workarounds for it).

It's still [failing](https://github.com/agilgur5/argo-workflows/actions/runs/8563028220/job/23467388104?pr=3) though, because it thinks it has something to `fetch` from the remote branch, even though it doesn't. So it needs some more work. 

Tbh, this is getting complicated enough that it perhaps should be split out into its own independent action

---

> It's still [failing](https://github.com/agilgur5/argo-workflows/actions/runs/8563028220/job/23467388104?pr=3) though, because it thinks it has something to `fetch` from the remote branch, even though it doesn't.

I fixed this by adding `fetch-depth` to `actions/checkout`. It passes now.... but the PR won't merge because the new commit doesn't run any checks, since GH won't run actions on changes from other actions (to avoid an infinite loop), per [previous comment](https://github.com/argoproj/argo-workflows/pull/12234#discussion_r1399762107).

So this whole approach isn't workable as checks will never run/pass on the final commit, meaning it will never be mergeable due to branch protections.

---

Reverted in https://github.com/argoproj/argo-workflows/pull/12892